src/pages/index.js

```diff
@@ -6,46 +6,83 @@ import PopupWithImage from "../components/PopupWithImage.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import Section from "../components/Section.js";
 import UserInfo from "../components/UserInfo.js";
+import ProfilePicEvt from "../components/ProfilePicEvt.js";
 import {
-  initialCards,
   config,
   editProfileButton,
+  editProfilePictureButton,
   addCardButton,
+  profilePicture,
 } from "../utils/constants.js";
+import { api } from "../components/Api.js";
+let gallery;
 //Add Card Popup Form
 const addCardPopupForm = new PopupWithForm("#add-card-modal", {
   handleSubmitEvent: (inputValues) => {
-    const newCard = {
-      name: inputValues.name,
-      link: inputValues.bio,
-    };
-    const cardElement = createCard(newCard);
-    gallery.addItem(`prepend`, cardElement);
+    addCardPopupForm.renderLoading(true);
+    api
+      .addCard(inputValues)
+      .then((inputValues) => {
+        const cardElement = createCard(inputValues);
+        gallery.addItem(`prepend`, cardElement);
+        addCardPopupForm.renderLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+        addCardPopupForm.renderLoading(false);
+      });
     addCardPopupForm.form.reset();
```

Comment:
`    addCardPopupForm.form.reset();
`

You need to **reset** the form only   in a `then` block in order not to do that immediately after `submit` because when there is an error, the modal will still be opened and the user can try to send the data again to the server without having to type the data into the inputs once more.